csharp_dotnetcore/16.Handling-Atachments/Handling_Attachments/AtachmentsBot.cs

```diff
@@ -0,0 +1,272 @@
+                case ActivityTypes.Message:
+
+                    // Take the input from the user and create the appropriate response.
+                    var reply = ProcessInput(turnContext);
+
+                    // Respond to the user.
+                    await turnContext.SendActivityAsync(reply, cancellationToken);
+
+                    await DisplayOptionsAsync(turnContext, cancellationToken);
+
+                    break;
+                case ActivityTypes.ConversationUpdate:
+                    // Send a welcome message to the user and tell them what actions they may perform to use this bot
+                    if (turnContext.Activity.MembersAdded.Any())
+                    {
+                        await SendWelcomeMessageAsync(turnContext, cancellationToken);
+                    }
+
+                    break;
+            }
+        }
+
+        private static async Task DisplayOptionsAsync(ITurnContext turnContext, CancellationToken cancellationToke)
+        {
+            var reply = turnContext.Activity.CreateReply();
+
+            // Create a HeroCard with options for the user to choose to interact with the bot.
+            var card = new HeroCard
+            {
+                Text = "You can upload an image or select one of the following choices",
+                Buttons = new List<CardAction>()
+                {
+                    // Note that some channel require different values to be used in order to get buttons to display text.
+                    // In this code the emulator is accounted for with 'title', but in other channels you may need to provide a value
+                    // for other parameters like 'text' or 'displayText'.
+                    new CardAction(ActionTypes.ImBack, title: "1. Inline Attachment", value: "1"),
+                    new CardAction(ActionTypes.ImBack, title: "2. Internet Attachment", value: "2"),
+                    new CardAction(ActionTypes.ImBack, title: "3. Uploaded Attachment", value: "3"),
+                },
+            };
+
+            // Add the card to our reply.
+            reply.Attachments = new List<Attachment>() { card.ToAttachment() };
+
+            await turnContext.SendActivityAsync(reply, cancellationToke);
+        }
+
+        /// <summary>
+        /// On a conversation update activity sent to the bot, the bot will
+        /// send a message to the any new user(s) that were added.
+        /// </summary>
+        /// <param name="turnContext">Provides the <see cref="ITurnContext"/> for the turn of the bot.</param>
+        /// <param name="cancellationToken" >(Optional) A <see cref="CancellationToken"/> that can be used by other objects
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>>A <see cref="Task"/> representing the operation result of the Turn operation.</returns>
+        private static async Task SendWelcomeMessageAsync(ITurnContext turnContext, CancellationToken cancellationToken)
```

Comment:
>SendWelcomeMessageAsync [](start = 34, length = 23)

This code is different than the greeting code we have in other bots. What's missing here?